akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -44,6 +44,16 @@ private[http] class HttpRequestRendererFactory(userAgentHeader: Option[headers.`
 
       def render(h: HttpHeader) = r ~~ h ~~ CrLf
 
+      val hostString: InetSocketAddress ⇒ String = {
+        // Retrieve the original host string that was given (IP or DNS name) if running Java 7 or later
+        // using the getHostString() method. This avoids a reverse DNS query from calling getHostName()
+        // if the original host string is an IP address.
+        val clazz = classOf[InetSocketAddress]
+        address ⇒ clazz.getMethods.collectFirst {
+          case m if m.getName == "getHostString" => m.invoke(address)
+        }.getOrElse(address.getHostName).asInstanceOf[String]
+      }
```

Comment:
>Yeah, my bad. Also would it be better to move hostString val to one level up - from onPush to HttpRequestRenderer or even to HttpRequestRendererFactory? Just to execute this as less times as possible.

That sounds like a fantastic idea.

> So here require is more for visibility and documentation 

It's primarily there to make sure we get the right `getHostString` method, if `getDeclaredMethod` or any of the `require` throws, then we'll fall back to using `getHostName`.

I considered falling back to `getHostName` whenever the reflective call fails, but I think that's the wrong way to go about it as it could fail for many reasons and it doesn't mean that the `getHostName` would succeed either. Makes sense?

Documenting why reflection is used is a very appropriate thing to do IMO.